### ISSUE TITLE
ParameterizedFunction changes

### ISSUE THREAD

**Issue by ChrisRackauckas** [korsbo/Latexify.jl#24]
Holding onto the SymEngine variables makes ParameterizedFunctions not compatible with precompilation so I plan on getting rid of them in the Julia v0.7 version. However, I wanted to make sure Latexify.jl could still work well. `funcs` is a `Vector{Expr}` so I am curious why you use the `Matrix{SymEngine.Basic}` instead of the `Matrix{Expr}` `Jex` for the Jacobian expression. If there's anything you need to ensure it's saved well we can make that change and drop the `symjac` field safely. Thanks for your work here!

Commit: https://github.com/JuliaDiffEq/ParameterizedFunctions.jl/commit/073ea741a9a42cb37dbbe7fe981498c4bb60201b

**Comment by korsbo** [comment]
I never actually use the `ode.symjac` field in the code, it is only used in an example in the documentation. 

I would certainly not mind using a pure expression based type, but I did not use the `Jex` field because it contained an expression (not a matrix) and it had `internal_var___` stuff in it. When double checking my facts right now, I realised that I am still using ParameterizedFunctions 3.0.0. Pkg.update is currently not my friend since it has unresolvable conflicts (probably related to the 0.7 update frenzy). Has `Jex` changed? 

Anyway, long story short, you will not break Latexify by removing the `symjac` field. We will just have to update the documentation. And if there is a field in your ode type which has a matrix of the jacobian, we will be able to render it.

**Comment by ChrisRackauckas** [comment]
Jex is the same and it's in there.

**Comment by korsbo** [comment]
```julia
odetest = @ode_def MyTest begin
    dx = y^3 - x
    dy = x*y - y*y
end

odetest.Jex
```
gives me:
```
quote 
    internal_var___J[1, 1] = -1 * 1
    internal_var___J[1, 2] = 3 * internal_var___u[2] ^ 2
    internal_var___J[2, 1] = internal_var___u[2] * 1
    internal_var___J[2, 2] = internal_var___u[1] - 2 * internal_var___u[2]
    nothing
end
```

I should however emphasise that my Julia packages are giving me grief at the moment, so my testing may be off.

**Comment by ChrisRackauckas** [comment]
Yup that's it. Can the name obfuscation be fixed or should I generate a different one?

**Comment by korsbo** [comment]
It can be fixed, either here or in ParameterizedFunctions. If ParameterizedFunctions would contain a field with something like
```julia
ode.jac_expr == [:(-1) :(3 * y^2); :(y) :(x-2*y)]
```
then `latexify(ode.jac_expr)` would work without needing to add some special parser for parameterizedFunctions. 

It would be possible to add support for `latexify(ode::AbstractParameterizedFunction; jacobian=true)` (or something) which gets the job done, but I would say that the first option may provide benefits to other projects, while the second is limited to only Latexify.

**Comment by ChrisRackauckas** [comment]
```julia
julia> f_t = @ode_def_nohes SymCheck begin # Checks for error due to symbol on 1
         dx = x
           dy = -c*y + d*x*y*t^2
           end a b c d
(::SymCheck{getfield(Main, Symbol("##6#12")),getfield(Main, Symbol("##7#13")),getfield(Main, Symbol("##8#14")),getfield(Main, Symbol("##9#15")),getfield(Main, Symbol("##10#16")),getfield(Main, Symbol("##11#17"))}) (generic function with 2 methods)

julia> f_t.symjac
2×2 Array{Any,2}:
 1                  0
  :(d * t ^ 2 * y)   :(-1c + d * t ^ 2 * x)

julia> f_t.symtgrad
2-element Array{Any,1}:
 0
  :(2 * d * t * x * y)
```

You can see https://github.com/symengine/SymEngine.jl/issues/122  in there but this is good enough and I'll fix the typing issue when that is closed.
